Add Debian "d-i" declaration to boot install

jbsky opened the pull request:
In order to do an automatic installation of a Debian distro via a PXE boot, it is necessary to declare a config file named :

```<tftp server>:///pxelinux.cfg/01-<mac address>.cfg``` 

Regard,

Jb

jbsky committed:
Add Debian "d-i" declaration to boot install

bertvv commented:
It seems that this configuration is always added, which doesn´t make sense in e.g. a network with EL machines. Can you add a flag (and document it) that ensures this declaration is only added when the user asks for it? Thanks!

jbsky commented:
Do you want it to be feature-flipped?

jbsky committed:
[fix] ansible.netcommon. missing

jbsky committed:
feat: add per-subnet UEFI PXE support (filename64) + fix pxeboot_server null guard

- Add subnet.filename64 support with architecture-based selection (option arch 00:07)
- Move 'option arch code 93' to global section (dhcp_global_pxe_arch)
- Fix dhcp_pxeboot_server guard: 'is defined and dhcp_pxeboot_server' to handle null values
- Allows per-subnet PXE config without global dhcp_pxeboot_server class override

jbsky committed:
fix: ipaddr filter crash when host IP not in subnet

When host.ip is not in a subnet, ipaddr(network) returns False.
Passing False to ipaddr('bool') causes 'Unrecognized type NoneType'.
Remove the redundant ipaddr('bool') - the truthiness of ipaddr(network)
result is sufficient.
